include/core/rtos/Mutex.hpp

```diff
@@ -0,0 +1,120 @@
+#ifndef EVT_RTOS_MUTEX_
+#define EVT_RTOS_MUTEX_
+
+#include <core/rtos/Initializable.hpp>
+
+namespace core::rtos {
+
+/**
+ * Class that wraps a Threadx Mutex. Mutexes are used to create mutual exclusion between threads.\n\n
+ *
+ * This class extends Initializable, and like all other Initializable classes must be passed into the
+ * Threadx::startKernel() method as part of the initList parameter.
+ */
+class Mutex : public Initializable {
+public:
+    /**
+     * Constructs a Mutex object, but does not initialize it (must call init before using).
+     *
+     * @param[in] name A pointer to the name of the Mutex.
+     * @param[in] priorityInheritance Whether or not threads currently holding this mutex should raise their priority
+     * to that of the highest priority thread waiting for the mutex.
+     */
+    Mutex(char* name, bool priorityInheritance);
+
+    TXError init(BytePoolBase& pool) override;
+
+    /**
+     * Mutex Destructor.
+     */
+    ~Mutex();
+
+    /**
+     * Attempts to obtain access to this Mutex. If the mutex is already in use by another thread,
+     * the calling thread will wait for waitOption ticks for the mutex to be obtained.
+     *
+     * @param waitOption How long (in ticks) the calling thread should wait for the mutex to be available.
+     * Use Enums::TXWait::WaitForever to wait forever.
+     * @return The first error found by the function (or Success if there was no error).
+     */
+    TXError acquire(uint32_t waitOption);
+
+    /**
+     * Releases control of the mutex, allowing other threads to acquire it.
+     *
+     * @return The first error found by the function (or Success if there was no error).
+     */
+    TXError release();
+
+    /**
+     * Prioritizes the list of waiting threads, placing the highest priority waiting thread at the front of the list.
+     * (Does not effect the order of the other waiting threads).
+     *
+     * @return The first error found by the function (or Success if there was no error).
+     */
+    TXError prioritize();
+
+    // Getters
+
+    /**
+     * Retrieves the name of this Mutex.
+     *
+     * @param[out] name a pointer to a place to store the name pointer.
+     * @return The first error found by the function (or Success if there was no error).
+     */
+    TXError getName(char** name);
```

Comment:
I agree that it's clunky, but I kind of feel that that's just how embedded code is. If you think it would be better, you could also have the `getName` function take a length, so the user specifies the maximum number of characters to copy over, cutting it short if the buffer is too small.

@mjh9585 do you know of a better way to handle this?

Reply:
I don't have a better way but I think passing in a string pointer and a maximum length is the best approach if it needs to be copied. I get the argument that the class and therefore the name could go out of scope before reading it, but the pointer approach is so clean and efficient. Also if you called this method then the class hasn't been deleted yet so maybe just putting a warning in the description to not store the pointer and use it immediately is enough. I am also thinking about the "Undefined behavior" which will at worst be reading random garbage from memory in the location where there was a string (no invalid memory address and statistically highly likely to hit \0 before end of memory) and I think in most cases will still get the name due to all of the objects being statically allocated before most of the stack operations. Likewise if this was in a thread's stack then that is still a ways away from the top of the stack due to the disjointed nature of having multiple stacks.

Reply:
I would prefer the pointer and max length approach here

Reply:
I implemented the pointer and max length approach.